modules/mesh-task/variables.tf

```diff
@@ -87,3 +89,15 @@ variable "tags" {
   type        = map(string)
   default     = {}
 }
+
+variable "tls" {
+  description = "Whether to enable TLS for the mesh-task."
```

Comment:
Maybe explain specifically this is for control plane traffic. Nit: not sure we have to refer to "the mesh-task" since this is a param for it so it feels slightly redundant.